commands/economy/buy.js

```diff
@@ -15,20 +15,24 @@ module.exports = {
   executeInteraction (client, locale, interaction) {
     if (interaction.database.economyEnabled !== 0) {
       if (interaction.options.getString('productname')) {
-        interaction.guild.locale = locale
-        fetchShopProduct(client, interaction.guild, interaction.options.getString('productname'), (productData) => {
-          if (productData) {
-            if (interaction.options.getString('properties')) productData.userInput = interaction.options.getString('properties').split(',')
-            buyItem(client, interaction.member, interaction.guild, productData, (status) => {
-              if (status.code) {
-                interaction.editReply({ embeds: [Success(status.ia || i18n(locale, 'BUY::SUCCESS', { ITEM: productData.productName }))] })
+        getMemberInventoryAndBalance(client, interaction.member, interaction.guild, (memberInventoryAndBalance) => {
+          getShopProduct(client, interaction.guild, interaction.options.getString('productname'), (shopProduct) => {
+            if (shopProduct) {
+              if (memberInventoryAndBalance >= shopProduct.productPrice) {
+                if (checkIfMemberHasProduct(client, interaction.member, interaction.guild, shopProduct.productId)) {
+                  interaction.editReply({ embeds: [Error(i18n(locale, 'BUY::ALREADYOWN'))] })
+                }
+                addItemToMemberInventory(memberInventoryAndBalance.inventory, shopProduct, (newInventory) => {
+                  updateMemberBalance(client, interaction.member, interaction.guild, (parseInt(memberInventoryAndBalance.amount) - shopProduct.productPrice))
+                  updateMemberInventory(client, interaction.member, interaction.guild, newInventory)
+                })
               } else {
-                interaction.editReply({ embeds: [Error(status.ia || i18n(locale, 'BUY::NOMONEY', { ITEM: productData.productName }))] })
+                interaction.editReply({ embeds: [Error(i18n(locale, 'BUY::NOMONEY', { ITEM: interaction.options.getString('productname') }))] })
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/alexandrumuntenas/pingu/pullRequest?prid=8657799)
```suggestion
                interaction.editReply({ embeds: [Error(i18n(locale, "BUY::NOMONEY", { ITEM: interaction.options.getString('productname') }))] })
```